pyathena/model.py

```diff
@@ -241,6 +241,158 @@ def result_reuse_minutes(self) -> Optional[int]:
         return self._result_reuse_minutes
 
 
+class AthenaCalculationExecutionStatus:
+    STATE_CREATING: str = "CREATING"
+    STATE_CREATED: str = "CREATED"
+    STATE_QUEUED: str = "QUEUED"
+    STATE_RUNNING: str = "RUNNING"
+    STATE_CANCELING: str = "CANCELING"
+    STATE_CANCELLED: str = "CANCELLED"
+    STATE_COMPLETED: str = "COMPLETED"
+    STATE_FAILED: str = "FAILED"
+
+    def __init__(self, response: Dict[str, Any]) -> None:
+        status = response.get("Status", None)
+        if not status:
+            raise DataError("KeyError `Status`")
+        self._state: Optional[str] = status.get("State", None)
+        self._state_change_reason: Optional[str] = status.get("StateChangeReason", None)
+        self._submission_date_time: Optional[datetime] = status.get("SubmissionDateTime", None)
+        self._completion_date_time: Optional[datetime] = status.get("CompletionDateTime", None)
+
+        statistics = response.get("Statistics", None)
+        if not statistics:
+            raise DataError("KeyError `Statistics`")
+        self._dpu_execution_in_millis: Optional[int] = statistics.get("DpuExecutionInMillis", None)
+        self._progress: Optional[str] = statistics.get("Progress", None)
+
+    @property
+    def state(self) -> Optional[str]:
+        return self._state
+
+    @property
+    def state_change_reason(self) -> Optional[str]:
+        return self._state_change_reason
+
+    @property
+    def submission_date_time(self) -> Optional[datetime]:
+        return self._submission_date_time
+
+    @property
+    def completion_date_time(self) -> Optional[datetime]:
+        return self._completion_date_time
+
+    @property
+    def dpu_execution_in_millis(self) -> Optional[int]:
+        return self._dpu_execution_in_millis
+
+    @property
+    def progress(self) -> Optional[str]:
+        return self._progress
+
+
+class AthenaCalculationExecution(AthenaCalculationExecutionStatus):
+    def __init__(self, response: Dict[str, Any]) -> None:
+        super(AthenaCalculationExecution, self).__init__(response)
+
+        self._calculation_id: Optional[str] = response.get("CalculationExecutionId", None)
+        if not self._calculation_id:
+            raise DataError("KeyError `CalculationExecutionId`")
```

Comment:
suggestion (non-blocking): Might be just a preference but if would be good to put this piece of logic into a function and call it like self.get_calculation_id(response.get("CalculationExecutionId")) and raise the DataError in that function

Reply:
I prefer to parse the value in the constructor of the current implementation, as it is harder to notice where the value is going wrong if there is logic in the get method.